src/iarc7_motion/iarc_tasks/go_to_roomba_task.py

```diff
@@ -0,0 +1,238 @@
+
+            elif (self._state == GoToObjectTaskState.init):
+                if self._roomba_array is None or self._drone_odometry is None:
+                    self._state = GoToObjectTaskState.waiting
+                else:
+                    self._state = GoToObjectTaskState.going
+                return (TaskRunning(), NopCommand())
+
+            elif (self._state == GoToObjectTaskState.waiting):
+                if self._roomba_array is None or self._drone_odometry is None:
+                    self._state = GoToObjectTaskState.waiting
+                else:
+                    self._state = GoToObjectTaskState.going
+                return (TaskRunning(), NopCommand())
+
+            elif self._state == GoToObjectTaskState.going:
+
+                if not (self._height_checker.above_min_maneuver_height(
+                            self._drone_odometry.pose.pose.position.z)):
+                    return (TaskAborted(msg='Drone is too low'),)
+                elif not (self._z_holder.check_z_error(
+                    self._drone_odometry.pose.pose.position.z)):
+                    return (TaskAborted(msg='Z error is too high'),)
+                elif not self._check_roomba_in_sight():
+                    return (TaskAborted(msg='The provided roomba is not in sight of quad'),)
+
+                try:
+                    roomba_transform = self._tf_buffer.lookup_transform(
+                                        'level_quad',
+                                        self._roomba_id,
+                                        rospy.Time.now(),
+                                        rospy.Duration(self._TRANSFORM_TIMEOUT))
+                except (tf2_ros.LookupException,
+                        tf2_ros.ConnectivityException,
+                        tf2_ros.ExtrapolationException) as ex:
+                    rospy.logerr('ObjectGoToTask: Exception when looking up transform')
+                    rospy.logerr(ex.message)
+                    return (TaskAborted(msg='Exception when looking up transform during go_to roomba'),)
+
+                # Creat point centered at drone's center
+                stamped_point = PointStamped()
+                stamped_point.point.x = 0
+                stamped_point.point.y = 0
+                stamped_point.point.z = 0
+
+                # returns point distances of roomba to center point of level quad
+                self._roomba_point = tf2_geometry_msgs.do_transform_point(
+                                                    stamped_point, roomba_transform)
+
+                """ This is for tracking
+                if not self._check_max_roomba_range():
+                    return (TaskAborted(msg='The provided roomba is not found or not close enough to the quad'),)
+                """
+
+                if self._check_max_ending_roomba_range():
+                    return (TaskDone(),)
+
+                roomba_x_velocity = self._roomba_odometry.twist.twist.linear.x
```

Comment:
Since the abstract will be passing in an ending radial distance away from the Roomba, lets not have any overshoot. So delete lines 161-166

Reply:
Fixed